cfme/infrastructure/datastore.py

```diff
@@ -343,6 +166,28 @@ def run_smartstate_analysis(self, wait_for_task_result=False):
             task.wait_for_finished()
             return task
 
+    def run_smartstate_analysis_from_provider(self, wait_for_task_result=False):
+        """ Runs smartstate analysis on this host
```

Comment:
[...] on this datastore

Reply:
below too, in the Note

Reply:
I updated the docstring the best I could.